flow/util/template_filters.py

```diff
@@ -109,6 +109,96 @@ def calc_tasks(operations, name, parallel=False, allow_mixed=False):
+    "g" for gigabytes, or "m" for megabytes.
+
+    Parameters
+    ----------
+    memory : str
+        Required memory to reserve per node.
+
+    Returns
+    -------
+    float
+        The parsed memory (numeric value) in gigabytes.
+    """
+    # This method is deprecated as of 0.13 and will be removed in 0.14.
+    try:
+        size_type = memory[-1]
+        if size_type.lower() == "m":
+            return float(memory[:-1]) / 1024
+        elif size_type.lower() == "g":
+            return float(memory[:-1])
+        else:
+            return float(memory)
+    except ValueError:
+        raise ValueError(
+            'Invalid memory passed. For gigabytes use suffix "g", '
+            'for megabytes use suffix "m", if a numeric value is passed then '
+            "it will be interpreted as memory in gigabytes."
+        )
+    except TypeError:
+        raise
+
+
+def check_memory(operations, memory=None):
+    """Check whether memory is defined by the user or not.
+
+    Parameters
+    ----------
+    operations : list
+        The operations of :class:`~._JobOperation` used to calculate the maximum memory required.
+    memory : str
+        Memory to reserve per node for all the operations. A valid memory flag passed to
+        `FlowProject.submit` should have a suffix of either "g" for gigabytes, or "m" for megabytes.
+
+    Returns
+    -------
+    bool
+        Evaluates to True if memory is defined by the user, else False.
+    """
+    return bool(memory) or any(
+        operation.directives["memory"] for operation in operations
+    )
+
+
+def calc_memory(operations, parallel=False, memory=None):
+    """Calculate the maximum memory to reserve for submission of operations.
+
+    Parameters
+    ----------
+    operations : list
+        The operations of :class:`~._JobOperation` used to calculate the maximum memory required.
+    parallel : bool
+        If True, operations are assumed to be executed in parallel, which means
+        that the total memory requested will be the sum of all memories requested instead of the
+        maximum memory requested. (Default value = False)
+    memory : str
+        Memory to reserve per node for all the operations. A valid memory flag passed to
+        `FlowProject.submit` should have a suffix of either "g" for gigabytes, or "m" for megabytes.
+
+    Returns
+    -------
+    float
+        The reserved memory (numeric value) per node in gigabytes.
+    """
+    if not memory:
+        list_of_memories = [
+            operation.directives["memory"] if operation.directives["memory"] else 0
+            for operation in operations
+        ]
+        if parallel:
+            return sum(list_of_memories)
+        else:
+            return max(list_of_memories)
+    else:
+        parsed_memory = _parse_memory_flag(memory)
+        return parsed_memory if not parallel else parsed_memory * len(operations)
```

Comment:
I don't think this logic is quite right. The `--memory` command-line argument's help says that this is the amount of memory to request *per node*. That is also likely why the old torque.sh template uses `pmem` instead of `mem`. Unfortunately, this complicates our logic quite a bit because we don't usually know how many nodes we will have a priori, and it's a fundamentally different calculation than the per operation calculation. Since the `--memory` flag is command line, and I suspect that very few people are using it anyway, I think we may be better off skipping a deprecation cycle for it and just removing it rather than trying to support two incompatible measures of memory.